cspp_runner/atms_rdr2sdr_runner.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023 Pytroll developers
+
+# Author(s):
+
+#   Adam Dybbroe <Firstname.Lastname@smhi.se>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Run the ATMS RDR to SDR processing withy CSPP on incoming DR data."""
+
+
+import socket
+from trollsift import Parser, globify
+import os
+import shutil
+
+import pathlib
+import tempfile
+from glob import glob
+from datetime import datetime
+
+import subprocess
+from urllib.parse import urlparse
+import logging
+import time
+
+import signal
+from queue import Empty
+from threading import Thread
+from posttroll.listener import ListenerContainer
+from posttroll.message import Message
+from posttroll.publisher import NoisyPublisher
+from cspp_runner.config import read_config
+from cspp_runner.runner import check_environment
+from cspp_runner.constants import (PLATFORM_SHORTNAMES,
+                                   PLATFORM_LONGNAMES)
+
+
+logger = logging.getLogger(__name__)
+
+
+class AtmsSdrRunner(Thread):
+    """The ATMS CSPP runner to process RDR to SDR files."""
+
+    def __init__(self, configfile):
+        """Initialize the runner object."""
+        super().__init__()
+
+        self.configfile = configfile
+        self.options = {}
+        config = read_config(self.configfile)
+        self.options = config
+
+        self.host = socket.gethostname()
+
+        self.sdr_file_patterns = self.options['sdr_file_patterns']
+        self._sdr_home = self.options['level1_home']
+
+        self.input_topics = self.options['subscribe_topics']
+        self.output_topics = self.options['publish_topics']
+
+        self._atms_sdr_call = self.options['atms_sdr_call']
+        self._atms_sdr_options = self.options['atms_sdr_options']
+
+        self.listener = None
+        self.publisher = None
+        self.loop = False
+        self._setup_and_start_communication()
+
+    def _setup_and_start_communication(self):
+        """Set up the Posttroll communication and start the publisher."""
+        logger.info("Starting up ATMS DR Runner")
+        logger.debug("Input topics:")
+        for top in self.input_topics:
+            logger.debug("{topic}".format(topic=str(top)))
```

Comment:
```suggestion
            logger.debug(str(top))
```

Unless I'm overlooking something, this can be simplified with no change in functionality.